src/gui/graphdesignerpage.h

```diff
@@ -25,19 +25,21 @@
 #include <QStringList>
 
 #include "core/include/enum.h"
+#include "core/mainapp.h"
+#include "core/graphinputs.h"
 
 #include "maingui.h"
 #include "abstractgraph.h"
 #include "attributerange.h"
-
-#include "core/mainapp.h"
-#include "core/graphinputs.h"
+#include "graphdesigner.h"
```

Comment:
recursive include:
`graphdesigner.h` includes `graphdesignerpage.h`; and `graphdesignerpage.h` includes `graphdesigner.h`

In any case, you already have a forward declaration of `GraphDesigner` at L41. So, you could remove L34 and add the include in the cpp file